### PR TITLE
Gradle 3.5 support

### LINKED ISSUE #112: Error with Gradle 3.4-rc-2: o.g.internal.component.Usage was renamed to UsageContext
Using `bintry-release` in a `com.android.library` project with Gradle 3.4-rc-2 fails with the following exception:

```
Caused by: java.lang.NoClassDefFoundError: org/gradle/api/internal/component/Usage
        at com.novoda.gradle.release.AndroidArtifacts.from(AndroidArtifacts.groovy:49)
```

The cause for this is that `Usage` was renamed to `UsageContext`.

To avoid such problems in the future, I created an issue to discuss public API for publishing in Gradle, as we seem to miss something in this area. Please add your thoughts on the topic here: https://github.com/gradle/gradle/issues/1361

### PR DESCRIPTION
**Pull request by passsy**
Fixes #112, #113 

Upgraded gradle to 3.4.1 but since 3.5 was just released I upgraded to the latest version.
Upgraded `AndroidLibrary.java` to support the latest internal API changes. 

---
I tried to solve it without using any internal APIs but AFAIK this would involve manual pom adjustments. This is all solved by the internal API.

### CODE REVIEW

**Comment by stefanhoth** [comment]
Can one of the admins review this PR?

File: core/src/main/groovy/com/novoda/gradle/release/AndroidLibrary.groovy [hunk 1]
```diff
@@ -5,13 +5,14 @@ import org.gradle.api.Project
...
+import org.gradle.api.attributes.Usage
 import org.gradle.api.internal.DefaultDomainObjectSet
 import org.gradle.api.internal.component.SoftwareComponentInternal
-import org.gradle.api.internal.component.Usage
+import org.gradle.api.internal.component.UsageContext
 
 class AndroidLibrary implements SoftwareComponentInternal {
 
-    private final Usage runtimeUsage
+    private final UsageContext runtimeUsage
```
Comment (tasomaniac) [inline core/src/main/groovy/com/novoda/gradle/release/AndroidLibrary.groovy]: Are these changes going to work if the users of the plugin have gradle version older than 3.5? I guess with this, the library will require the users to use version 3.4?

Comment (passsy) [inline core/src/main/groovy/com/novoda/gradle/release/AndroidLibrary.groovy]: yes, it doesn't work with 3.3

**Comment by stefanhoth** [comment]
Why is the CI still using my token. 😡

File: core/src/main/groovy/com/novoda/gradle/release/AndroidLibrary.groovy (hunk 1 above)
Comment (tasomaniac) [inline core/src/main/groovy/com/novoda/gradle/release/AndroidLibrary.groovy]: I guess it is fine as long as we make it clear on the release notes.

Comment (mr-archano) [inline core/src/main/groovy/com/novoda/gradle/release/AndroidLibrary.groovy]: @tasomaniac I just update issue #112. I don't think we have other option than re-release for 3.4+ only. More details can be found [here](https://github.com/novoda/bintray-release/issues/112#issuecomment-293674149).

**Comment by mr-archano** [comment]
@stefanhoth it seems like a glitch. Apparently PRB has not been triggered at all.

**Comment by stefanhoth** [comment]
Retest this please.

**Comment by stefanhoth** [comment]
Ok to test.

**Comment by stefanhoth** [comment]
Ok that was it. Sorry for the spam and thank you for the contribution @passsy!

**Comment by ouchadam** [comment]
@mr-archano do you think it's worth throwing in the code if the version of gradle isn't 3.4+ or will a warning in the readme be enough?

**Comment by tasomaniac** [comment]
@ouchadam it will fail anyways with mysterious error output. If we can detect the version of the gradle used, we can fail fast and tell them to update their gradle. Probably this will happen when they update our plugin. So it will be easier for them to update gradle too.

**Comment by passsy** [comment]
I'm absolutely for the gradle version detection with a good error message

**Pull request merged**